pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -781,12 +776,191 @@ podCopyPhase:
 	}
 }
 
+// getInstallSlotForPod tries to allocate a local scoped install slot for a pod
+func getInstallSlotForPod(installTracker []chan struct{}, podName string) bool {
+	scopedLog := log.WithName("getInstallSlotForPod")
+	podID, err := getOrdinalValFromPodName(podName)
+	if err != nil {
+		scopedLog.Error(err, "unable to derive podId for podname", podName)
+		return false
+	}
+
+	select {
+	case installTracker[podID] <- struct{}{}:
+		return true
+	default:
+		return false
+	}
+}
+
+// freeInstallSlotForPod frees up an install slot for a pod
+func freeInstallSlotForPod(installTracker []chan struct{}, podName string) {
+	scopedLog := log.WithName("freeInstallSlotForPod")
+	podID, err := getOrdinalValFromPodName(podName)
+	if err != nil {
+		scopedLog.Error(err, "unable to derive podId for podname", podName)
+		return
+	}
+
+	select {
+	case <-installTracker[podID]:
+	default:
+		scopedLog.Error(nil, "trying to free an install slot without even allocating it")
+	}
+}
+
+// isPendingClusterScopeWork confirms if there is any pending custer scoped app work
+func isPendingClusterScopeWork(afwPipeline *AppInstallPipeline) bool {
+	// CR doesn't deal with cluster scoped apps
+	if !isClusterScoped(afwPipeline.cr.GetObjectKind().GroupVersionKind().Kind) {
+		return false
+	}
+
+	// There is no cluster scoped apps pending for bundle push
+	if afwPipeline.appDeployContext.BundlePushStatus.BundlePushStage == enterpriseApi.BundlePushComplete || afwPipeline.appDeployContext.BundlePushStatus.BundlePushStage == enterpriseApi.BundlePushUninitialized {
```

Comment:
I think the last time we discussed was about the explicit status value, that we planned to set when we detect cluster scoped apps every time.  Here this is needed when the CR is configured with just the local scoped apps. Otherwise, it will cause an unnecessary bundle push. Here, are just checking for the value, otherwise, we are not setting explicitly anywhere.

Reply:
If we are not setting it anywhere then better to just check `afwPipeline.appDeployContext.BundlePushStatus.BundlePushStage < enterpriseApi.BundlePushPending`. What do you think?

Reply:
Ya, that also works. Here we are checking with `BundlePushUninitialized `, so we are good, which makes it readable :-)